### DISCUSSION TITLE
The value passed in the previous view correspond to either state or environment?

### DISCUSSION THREAD

**Discussion by ghost** [pointfreeco/swift-composable-architecture#1271] [Q&A]
If there is a list view and a detail view, let's assume that id is needed for detail information in the detail view.
If so, does the id of the detail view belong to the state? Or does it belong to the environment?

I think this value is the environment, because regardless of the state of the detail view. 
And this value just passed from list view. am I wrong?

I saw the example code here, but I couldn't figure it out because it was made up of a single view.

**Comment by mbrandonw** [comment]
Hi @jay-linger, I made need to see some sample code to fully understand, but typically you would model this as having some detail state that contains an ID:

```swift
struct DetailState: Identifiable {
  let id: UUID
  …
}
```

And then your feature holds onto an identified array of that state:

```swift
struct FeatureState {
  var rows: IdentifiedArrayOf<DetailState>
}
```

In this case both the list feature and the detail have access to the ID.

I also do not think that putting the ID in the environment is appropriate. The environment is best suited for holding dependencies that need to interact with the outside world, and that does not seem to be the situation here.

**Reply by ghost** [reply]
@mbrandonw Thank you for reply.

Sorry about that no sample code.

So, let's see code.

```swift
// Model
struct Todo {
    let id: Int
    let title: String
    let date: String
}

// List State
struct ListState {
    var todos = [Todo]()
    ...
}
```

Than at detail view need specific Todo's id for detail API.
So I think like that.

```swift
struct DetailEnvironment {
    var id: Int
}

// reducer
...
case .onAppear:
    return .task { detailAPI(id: environment.id }
...
```

There are parts that don't fit the syntax because I short it, but I think you can understand well.

But if the environment is related to the environment outside of the world, as you said, state would be the right place.
